tools/gen-intrinsics-specs.py

```diff
@@ -93,51 +94,86 @@ def rst_literal_quote(mapping, workflow):
         return '```\n' + '\n'.join(lines) + '\n```'
 
 
-def quote_split_intrinsics(intrinsic, workflow):
+def quote_split_intrinsics(intrinsic, workflow, baseurl=__ARMDEVELOPER):
     r"""
     >>> quote_split_intrinsics('int f(int x, float y)', 'rst')
     '.. code:: c\n\n    int f(\n        int x,\n        float y)'
 
     >>> quote_split_intrinsics('int f(int x)', 'rst')
     '.. code:: c\n\n    int f(int x)\n'
 
-    >>> quote_split_intrinsics('int f(int x, float y)', 'markdown')
-    '`int f(`<br>&nbsp;&nbsp;&nbsp;&nbsp;`int x,`<br>&nbsp;&nbsp;&nbsp;&nbsp;` float y)`'
+    >>> quote_split_intrinsics('int f(int x, float y)', 'markdown', 'baseurl/')
+    '<code>int <a href="baseurl/f" target="_blank">f</a>(<br>&nbsp;&nbsp;&nbsp;&nbsp; int x,<br>&nbsp;&nbsp;&nbsp;&nbsp; float y)</code>'
 
-    >>> quote_split_intrinsics('int f(int x)', 'markdown')
-    '`int f(int x)`'
+    >>> quote_split_intrinsics('int f(int x)', 'markdown', 'baseurl/')
+    '<code>int <a href="baseurl/f" target="_blank">f</a>(int x)</code>'
 
-    >>> quote_split_intrinsics('int f(int x, float y)', 'pdf')
+    >>> quote_split_intrinsics('int f(int x, float y)', 'pdf', 'baseurl/')
     '``` c\nint f(\n  int x,\n  float y)\n```'
 
-    >>> quote_split_intrinsics('int f(int x)', 'pdf')
+    >>> quote_split_intrinsics('int f(int x)', 'pdf', 'baseurl/')
     '`int f(int x)`'
     """
-    # Remove the suffix ')' from the intrinsic string.
-    intrinsic_without_ending = intrinsic[:-1]
-    ret_def, par, signature = intrinsic_without_ending.partition('(')
-    split_signature = signature.split(',')
+
     whitespace_indent = "&nbsp;&nbsp;&nbsp;&nbsp;"
 
-    if len(split_signature) > 1:
+    intrinsic_signature = get_signature(intrinsic)
+    intrinsic_link_id = get_intrinsic_name(intrinsic)
+    intrinsic_type = get_intrinsic_return_type(intrinsic)
+
+    # This replacement has been added in order to avoid issues with links
+    # that contain square brackets.
+    # See also: https://stackoverflow.com/questions/13013987/ruby-how-to-escape-url-with-square-brackets-and/17901435
+    formatted_intrinsic_id = intrinsic_link_id.replace("[","%5B").replace("]","%5D")
+    formatted_site_link = f"{baseurl}{formatted_intrinsic_id}"
```

Comment:
I think we should use a library to join the two pieces of the URL, so that even if we forget to specify the `/` token between the two pieces of the URL, it gets added automatically. For example, the following four situation should end up returning the same URL `baseurl/A`:

1. `baseurl/` + `A`
2. `baseurl` + `A`
3. `baseurl` + `/A`
4. `baseur/` + `/A`

You can use the library `urllib` to do that, see the first example in https://www.codegrepper.com/code-examples/python/python+url+join

Once you have modified that, please replace all the `baseurl/` input string in the unit tests with just `baseurl`.